discovery-infra/tests/test_proxy.py

```diff
@@ -1,58 +1,79 @@
+import os
 import pytest
 import openshift as oc
-from logger import log
+import logging
 
 from tests.base_test import BaseTest
 from tests.conftest import env_variables
+from test_infra.controllers.proxy_controller.proxy_controller import ProxyController
 
-
-@pytest.mark.skipif(not env_variables['http_proxy_url'], reason="no proxy environment")
 class TestProxy(BaseTest):
+    @pytest.fixture()
+    def proxy_server(self):
+        logging.info('--- SETUP --- proxy controller')
+        proxy_servers = []
+
+        def start_proxy_server(**kwargs):
```

Comment:
not sure i understand what is the reason to have it as internal function?

Reply:
This fixture is a generator so that I can start more than one server for the tests. For example one for http and one for https

Reply:
https://github.com/nshidlin/assisted-test-infra/blob/add-proxy-controller/discovery-infra/tests/test_proxy.py#L51